### PR TITLE
amazon ec2 support for machine

### PR DESCRIPTION
**Pull request by will-chan**

### CODE REVIEW

File: code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            "type": "string",
+            "nullable": false,
+            "required": true
+        },
+        "vpcId": {
+            "type": "string",
+            "nullable": false,
+            "required": true
+        },
+        "sessionToken": {
```
Comment (cjellick) [inline code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json]: Fields that cant be null (`nullable=false`) should be required (`required=trued`).
Or fields that are not required (`required=false`) should be nullable (`nullable=true`).

...I think

File: code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        },
+        "securityGroup": {
+            "type": "string",
+            "nullable": false
+        },
+        "instanceType": {
+            "type": "string",
+            "nullable": false
+        },
+        "rootSize": {
```
Comment (cjellick) [inline code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json]: rootSize could be an integer.

**Comment by cjellick** [comment]
Ping @will-chan. Done reviewing. A couple minor comments

File: code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json (hunk 2 above)
Comment (cjellick) [inline code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json]: On second thought, probably better to just leave as a string.

**Comment by cjellick** [comment]
@alena1108 or @sonchang can one of you also review?

File: code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json (hunk 1 above)
Comment (will-chan) [inline code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json]: yep, cut and paste error.

**Comment by will-chan** [comment]
updated the json files as well with latest comments.

**Comment by alena1108** [comment]
LGTM

**Comment by cjellick** [comment]
LGTM

**Pull request merged**